src/context.cpp

```diff
@@ -485,25 +491,25 @@ namespace librealsense
 
             //Assumes dds_device initialization finished
             size_t count = _dds_dev->foreach_sensor(
-                [&]( size_t sensor_index, const std::string & name ) {
-                    this->add_dds_sensor( _dds_dev, sensor_index, name );
+                [&]( const std::string & name ) {
+                    this->add_dds_sensor( _dds_dev, name );
                 } );
             for( size_t i = 0; i < count; ++i )
             {
-                software_sensor & sensor = get_software_sensor( i );
-                _dds_dev->foreach_video_profile( i, [&]( const rs2_video_stream& profile, bool default_profile ) {
+                dds_sensor_proxy & sensor = reinterpret_cast<dds_sensor_proxy &>(get_software_sensor( i ));
```

Comment:
Please explain why we must have here a `reinterpret_cast`

Reply:
👍 
There is a rule I once read:
```
There should be a really good reason why to use `reinterpret_cast`.
You shouldn't use it, if you must you need to add a comment why :)
```

Not sure I always did it, but it's good to think of it when you add it.